### ISSUE TITLE
IDE-agnostic way of generating code.

### ISSUE THREAD

**Issue by roughiain** [mrlacey/XamlStyleTypes#1]
Allow the functionality to be decoupled from the IDE so users can use in VSCode/Rider.

**Comment by mrlacey** [comment]
I'm making what feels like good progress on this.

![image](https://user-images.githubusercontent.com/189547/235363665-ba915b27-da63-4f31-81b1-6eabe1bdb6ba.png)

Two things I don't think I can work around:
- Won't get nesting of the generated files
- Have to specify the namespace to use for generation 😢 


But this should give support to
- VSCode
- VS for Mac
- Rider


testing continues...

**Comment by mrlacey** [comment]
It looks like it'll be possible to do this by just adding a NuGet package to a project. If that was (is) possible, is it worth keeping the Visual Studio extension (custom tool) option available as well?

**Comment by roughiain** [comment]
Sounds amazing being able to have a NuGet package to do this. Personally I have never used many custom tools in VS so I'm not sure how widely they are used. I would guess having it as a NuGet package would be the more popular option...

**Comment by mrlacey** [comment]
Woot! This is looking good.

Here it is working in Rider.

All I did was add the NuGet package reference.

![image](https://github.com/mrlacey/XamlStyleTypes/assets/189547/a2651083-ae49-4677-b027-ac15f2f5fe18)

_Specifying a namespace is now optional too._


It still needs a bit more testing (in VSCode) and tidying up before a proper release. Getting close though...

**Comment by mrlacey** [comment]
Right, I need to stop putting this off.
It works in VSCode too!

I just added the package and Bam! generated files.
It doesn't seem as quick as updating the generated files (I had to build to force a regen) but I can live with that.

![image](https://github.com/mrlacey/XamlStyleTypes/assets/189547/beec32aa-9584-4895-8e84-d83b069c60dd)

I could get my Mac out and try it in VS4Mac but, for now, I'm going to assume it will work there too.

Now to package this thing appropriately
